src/withResolution/mediaQuery.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+type Callback = (mql: MediaQueryList, alias: string) => void
+type QueryData = {
+  [query: string]: {|
+    +mediaQuery: MediaQueryList,
+    +allCallbacks: (mql: MediaQueryListEvent) => void,
+  |},
+}
+type PerCallbackData = {
+  [query: string]: Array<{|
+    +alias: string,
+    +callback: Callback,
+  |}>,
+}
+
+const perCallbackData: PerCallbackData = {}
+const queryData: QueryData = {}
+
+function callCallbacks(query) {
+  return () => {
+    perCallbackData[query].forEach(({ callback, alias }) =>
+      callback(queryData[query].mediaQuery, alias)
+    )
+  }
+}
+
+export const supportsMatchMedia = !!window.matchMedia
```

Comment:
Could this be more robust? something like `window && typeof window.matchMedia === 'function'`

Mostly just thinking about the React Native use case where `window` will not be available.

Although the more I think about it, the more I think that RN will need its own implementation of `withResolution` since most of the logic happening in this module will not really work in a RN environment

Reply:
it's still a good point, we shouldn't assume window, will update